### PR TITLE
test: add e2e tests with playwright

### PR DESCRIPTION
**Pull request by msebastianb**
**Changes**
- Add and configure playwright to the project
- Add command to run tests
- Implement some tests for current features

Closes https://github.com/fastify/fastify-swagger-ui/issues/98

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -150,3 +150,9 @@ yarn.lock
 # editor files
 .vscode
 .idea
+/test-results/
+/playwright-report/
+/playwright/.cache/
+/test-results/
+/playwright-report/
+/playwright/.cache/
```
Comment (Fdawgs) [inline .gitignore:158]: Duplicates

File: playwright.config.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const { defineConfig, devices } = require('@playwright/test')
```
Comment (Fdawgs) [inline playwright.config.js:1]: Missing use strict directive

File: .github/workflows/playwright.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+name: Playwright Tests
+on:
+  push:
+    branches: [ main, master ]
+  pull_request:
+    branches: [ main, master ]
```
Comment (Fdawgs) [inline .github/workflows/playwright.yml:6]: Should follow the same triggers as the main ci workflow

File: .github/workflows/playwright.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+jobs:
+  test:
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
```
Comment (Fdawgs) [inline .github/workflows/playwright.yml:15]: use `lts/*`

File: .github/workflows/playwright.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    - name: Install Playwright Browsers
+      run: npx playwright install chromium --with-deps
+    - name: Run Playwright tests
+      run: npx playwright test
+    - uses: actions/upload-artifact@v3
+      if: always()
+      with:
+        name: playwright-report
+        path: playwright-report/
+        retention-days: 30
```
Comment (Fdawgs) [inline .github/workflows/playwright.yml:27]: What would we use this for?

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: .github/workflows/playwright.yml (hunk 5 above)
Comment (msebastianb) [reply .github/workflows/playwright.yml:27]: It may not be useful here.

**Comment by Uzlopak** [comment]
Hmm. problem is, that automerge is in the ci.yml and playwright is not taken into account. So it could result in merging a PR despite the playwright wf failed.

File: .github/workflows/playwright.yml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: lts/*
+    - name: Install dependencies
+      run: npm i
+    - name: Install Playwright Browsers
+      run: npx playwright install chromium --with-deps
```
Comment (Eomm) [inline .github/workflows/playwright.yml:30]:
```suggestion
      run: npx playwright@1 install chromium --with-deps
```

File: .github/workflows/playwright.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    - name: Run Playwright tests
+      run: npx playwright test
```
Comment (Eomm) [inline .github/workflows/playwright.yml:32]:
```suggestion
      run: npx playwright@1 test
```

**Review by Eomm — COMMENTED** [review]
> Hmm. problem is, that automerge is in the ci.yml and playwright is not taken into account. So it could result in merging a PR despite the playwright wf failed.

I think we could merge this new job in the existing one:
https://github.com/fastify/fastify-swagger-ui/blob/master/.github/workflows/ci.yml

then we can run playwright first if I'm not wrong

**Comment by Uzlopak** [comment]
maybe put everything into the test script of package.json/npm.

**Review by Uzlopak — APPROVED** [review]
Its odd because the integration tests should run after the unit tests, but it works for me...

But lets see what the others say.

**Comment by Fdawgs** [comment]
CI isn't happy :/

**Comment by msebastianb** [comment]
@Fdawgs WDYM?

**Comment by Fdawgs** [comment]
> @Fdawgs WDYM?

It's happy now, don't worry 😊

File: .github/workflows/playwright.yml [hunk 8]
```diff
@@ -0,0 +1,21 @@
+name: Playwright Tests
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
```
Comment (Fdawgs) [inline .github/workflows/playwright.yml:7]: Can you declare the job permissions please? Last change I promise :D

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**